packages/cli/src/commands/build/index.ts

```diff
@@ -820,60 +813,51 @@ function mergeWildcard(
   return wildcard;
 }
 
-function mergeFlags(
-  buildResults: Iterable<BuildResult | BuildOutputConfig>
-): BuildResultV2Typical['flags'] {
-  return Array.from(buildResults).flatMap(result => {
-    if ('flags' in result) {
-      return result.flags ?? [];
-    }
-
-    return [];
-  });
-}
-
 /**
- * Takes the build output and writes all the variants into the `variants.json`
- * file. It'll skip variants that already exist.
+ * Takes the build output and writes all the flags into the `flags.json`
+ * file. It'll skip flags that already exist.
  */
-async function writeVariantsJson(
+async function writeFlagsJSON(
   { output }: Client,
   buildResults: Iterable<BuildResult | BuildOutputConfig>,
   outputDir: string
 ): Promise<void> {
-  const variantsFilePath = join(outputDir, 'variants.json');
+  const flagsFilePath = join(outputDir, 'flags.json');
```

Comment:
<!-- Convention comments refer to a team, repo, or project convention that is not being followed. Mention the convention and how it has been violated.  -->
**convention (blocking):** Can we add a test that this file is written under the correct circumstances?‏